In [1]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import statsmodels.api as sm
import statistics as stat
from sklearn.linear_model import LinearRegression 
import math

In [2]:
datafile = '../../dataset/ch07/bookstore_sales.txt'
bookstore = pd.read_csv(datafile, sep='\t')

In [3]:
bookstore.info

<bound method DataFrame.info of    Number of Sales People Working  Sales (in $1000)
0                               2                10
1                               3                11
2                               7                13
3                               9                14
4                              10                18
5                              10                20
6                              12                20
7                              15                22
8                              16                22
9                              20                26>

In [4]:
len(bookstore)

10

In [5]:
bookstore.columns

Index(['Number of Sales People Working', 'Sales (in $1000)'], dtype='object')

In [6]:
bookstore.head()

Number of Sales People Working  Sales (in $1000)
0                               2                10
1                               3                11
2                               7                13
3                               9                14
4                              10                18

*Prolem 5*


a) Find the slope estimate, b1.

b) What does it mean, in this context?

c) Find the intercept, b0.

d) What does it mean, in this context? Is it meaningful?

e) Write down the equation that predicts Sales from Number of
Sales People Working.

f) If 18 people are working, what Sales do you predict?

g) If sales are actually $25,000, what is the value of the residual?

h) Have we overestimated or underestimated the sales?

In [7]:
## a) Find the slope estimate, b1.

r = 0.965
sd_y = 5.34
sd_x = 5.64
b_1 = sd_y / sd_x * r
round (b_1, 3)

0.914

** b) What does it mean, in this context?


Answer: The slope, 0.914, says that a sales of 1000 dollars with one more sales person working can be expected, on average, to have 0.914 more of 1000 dollars sales.


In [8]:
## c) Find the intercept, b0.

y_bar = 17.6
x_bar = 10.4
r = 0.965
sd_y = 5.34
sd_x = 5.64
b_1 = sd_y / sd_x * r
b_0 = y_bar - b_1 * x_bar
round (b_0, 3)

8.098

** d) What does it mean, in this context? Is it meaningful?


Answer: The intercept, 8.098, says that the model predicts that even a sales with no sales people would have, on average, about 8.090 of dollars of sales.  No, the intercept is not meaningful here, because the intercept serves only as a starting value for prediction.

** e) Write down the equation that predicts Sales from Number of Sales People Working.


Answer: y_hat = b0 + b1x

In [9]:
## f) If 18 people are working, what Sales do you predict?

y_bar = 17.6
x_bar = 10.4
r = 0.965
sd_y = 5.34
sd_x = 5.64
b_1 = sd_y / sd_x * r
b_0 = y_bar - b_1 * x_bar

r = 0.965
sd_y = 5.34
sd_x = 5.64
b_1 = sd_y / sd_x * r

x = 18

y_hat = b_0 + b_1 * x

print (f"If 18 people are working, I predict {y_hat:.3f} of sales.")

If 18 people are working, I predict 24.544 of sales.


In [10]:
## g) If sales are actually $25,000, what is the value of the residual?

y_observied_value = 25000


In [11]:
## h) Have we overestimated or underestimated the sales?

r = 0.965
r_2 = 0.965 * 0.965
round (r_2, 2)

0.93

In [12]:
## h)
r_2 = 0.93

r_error_term = 1 - (r_2)

print (f"The error term is {r_error_term*100:.2f} precent")



The error term is 7.00 precent


** h)


Answer: So 7 precent of the variability in toal sales has been left in the residuals, therefore, there is 93 precent of association between sales and number of sales people working.

** Problem 9


a) What are the units of the residuals?

b) Which residual contributes the most to the sum that was
minimized according to the least squares criterion to find
this regression?

c) Which residual contributes least to that sum?

* a) What are the units of the residuals?


Answer: The units of the residuals are in thousands of dollars.

* b)Which residual contributes the most to the sum that was
minimized according to the least squares criterion to find
this regression?


Answer: 2.77 residual contributes the most to the sum that was minimized according to the least squares criterion to find this regression.

* c) Which residual contributes least to that sum?


Answer: 0.07 residual contributes the least to that sum.

** 11 For the regression model for the bookstore of Exercise 5, what is the value of R2 and what does it mean?


Answer: From the regression model for the bookstore of Exercise 5, the R2 value is 0.93 and it means there is 93 precent of association between sales and number of sales people working.

** 27. Tell what each of the residual plots below indicates about the appropriateness of the linear model that was fit to the data.


Answer: The residual plot (b) shows the appropriateness of the linear model that was fit to the data, because the plot (b) does not have non-linear signal, no obvious outliers , and the plot does not thicken.

** 28 Tell what each of the residual plots below indicates about the appropriateness of the linear model that was fit to the data.


Answer: The residual plot (a) shows the appropriateness of the linear model that was fit to the data, because the plot (a) does not have obvious outliers, the plot does not thicken and no non-linear signal.
